mergify_engine/delayed_refresh.py

```diff
@@ -0,0 +1,86 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2021 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import datetime
+import typing
+
+import daiquiri
+
+from mergify_engine import context
+from mergify_engine import github_types
+from mergify_engine import rules
+from mergify_engine import utils
+from mergify_engine import worker
+from mergify_engine.rules import filter
+
+
+LOG = daiquiri.getLogger(__name__)
+
+DELAYED_REFRESH_KEY = "delayed-refresh"
+
+
+async def plan_next_refresh(match: rules.RulesEvaluator, ctxt: context.Context) -> None:
+    best_bet: typing.Optional[datetime.datetime] = None
+    for rule in match.matching_rules:
+        f = filter.NearDatetimeFilter(rule.conditions.extract_raw_filter_tree())
+        bet = await f(ctxt.pull_request)
+        if best_bet is None or best_bet > bet:
+            best_bet = bet
+
+    zset_subkey = f"{ctxt.repository.installation.owner_id}~{ctxt.repository.installation.owner_login}~{ctxt.repository.repo['id']}~{ctxt.repository.repo['name']}~{ctxt.pull['number']}"
```

Comment:
why do we include the login here?
I'd think sticking to `id` was safer ane enough here.

Reply:
Ok, I'm fine with it. I just wanted to be sure it was an explicit choice.
I imagine it'd be ok to remove some name-based logging in the worker since this is not useful in general on a day-to-day basis.

Then the org-renaming issue would be "solved". But that can also be done later.